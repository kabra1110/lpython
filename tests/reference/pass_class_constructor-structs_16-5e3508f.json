{
    "basename": "pass_class_constructor-structs_16-5e3508f",
    "cmd": "lpython --cumulative --pass=class_constructor --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/structs_16.py",
    "infile_hash": "e87ea0ba10cb7fcd6ce2eb6d2e953a26dcd6a843f6cf4e891287dcb1",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "pass_class_constructor-structs_16-5e3508f.stdout",
    "stdout_hash": "355b37762d2ce7b10433387aef367b7348cf9e0f87620e2e21148527",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}