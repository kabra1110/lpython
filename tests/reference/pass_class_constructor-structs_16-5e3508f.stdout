(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        9
                        {
                            A:
                                (StructType
                                    (SymbolTable
                                        2
                                        {
                                            B:
                                                (UnionType
                                                    (SymbolTable
                                                        3
                                                        {
                                                            x:
                                                                (Variable
                                                                    3
                                                                    x
                                                                    []
                                                                    Local
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 4)
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                ),
                                                            y:
                                                                (Variable
                                                                    3
                                                                    y
                                                                    []
                                                                    Local
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Integer 8)
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                )
                                                        })
                                                    B
                                                    []
                                                    [x
                                                    y]
                                                    Source
                                                    Public
                                                    [(())
                                                    (())]
                                                    ()
                                                ),
                                            b:
                                                (Variable
                                                    2
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Union
                                                        2 B
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            c:
                                                (Variable
                                                    2
                                                    c
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    A
                                    []
                                    [b
                                    c]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    [(())
                                    (())]
                                    ()
                                    ()
                                ),
                            _lpython_main_program:
                                (Function
                                    (SymbolTable
                                        8
                                        {
                                            
                                        })
                                    _lpython_main_program
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [test_ordering]
                                    []
                                    [(SubroutineCall
                                        9 test_ordering
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            test_ordering:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            1_A_b:
                                                (ExternalSymbol
                                                    4
                                                    1_A_b
                                                    2 b
                                                    A
                                                    []
                                                    b
                                                    Public
                                                ),
                                            1_A_c:
                                                (ExternalSymbol
                                                    4
                                                    1_A_c
                                                    2 c
                                                    A
                                                    []
                                                    c
                                                    Public
                                                ),
                                            A_B:
                                                (ExternalSymbol
                                                    4
                                                    A_B
                                                    2 B
                                                    A
                                                    []
                                                    B
                                                    Public
                                                ),
                                            ad:
                                                (Variable
                                                    4
                                                    ad
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        9 A
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            bd:
                                                (Variable
                                                    4
                                                    bd
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Union
                                                        4 A_B
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    test_ordering
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(=
                                        (Var 4 bd)
                                        (UnionTypeConstructor
                                            4 A_B
                                            []
                                            (Union
                                                4 A_B
                                            )
                                            ()
                                        )
                                        ()
                                    )
                                    (=
                                        (UnionInstanceMember
                                            (Var 4 bd)
                                            3 x
                                            (Integer 4)
                                            ()
                                        )
                                        (IntegerConstant 1 (Integer 4))
                                        ()
                                    )
                                    (=
                                        (StructInstanceMember
                                            (Var 4 ad)
                                            4 1_A_b
                                            (Union
                                                2 B
                                            )
                                            ()
                                        )
                                        (Var 4 bd)
                                        ()
                                    )
                                    (=
                                        (StructInstanceMember
                                            (Var 4 ad)
                                            4 1_A_c
                                            (Integer 4)
                                            ()
                                        )
                                        (IntegerConstant 2 (Integer 4))
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (UnionInstanceMember
                                                (StructInstanceMember
                                                    (Var 4 ad)
                                                    2 b
                                                    (Union
                                                        2 B
                                                    )
                                                    ()
                                                )
                                                3 x
                                                (Integer 4)
                                                ()
                                            )
                                            Eq
                                            (IntegerConstant 1 (Integer 4))
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (StructInstanceMember
                                                (Var 4 ad)
                                                2 c
                                                (Integer 4)
                                                ()
                                            )
                                            Eq
                                            (IntegerConstant 2 (Integer 4))
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    _global_symbols
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        7
                        {
                            _lpython_main_program:
                                (ExternalSymbol
                                    7
                                    _lpython_main_program
                                    9 _lpython_main_program
                                    _global_symbols
                                    []
                                    _lpython_main_program
                                    Public
                                )
                        })
                    main_program
                    [_global_symbols]
                    [(SubroutineCall
                        7 _lpython_main_program
                        ()
                        []
                        ()
                    )]
                )
        })
    []
)
